battle-engine.js

```diff
@@ -149,7 +149,20 @@ BattlePokemon = (function () {
 		this.level = this.battle.clampIntRange(set.forcedLevel || set.level || 100, 1, 9999);
 
 		let genders = {M:'M', F:'F'};
-		this.gender = this.template.gender || genders[set.gender] || (Math.random() * 2 < 1 ? 'M' : 'F');
+		let rulesetContainsIllegal = false;
+		let format = this.battle.getFormat();
+		if (format && format.ruleset) {
+			for (let i = 0; i < format.ruleset.length; i++) {
+				if (format.ruleset[i].indexOf('Standard') > -1) {
+					rulesetContainsIllegal = true;
+				}
+			}
+		}
```

Comment:
Honestly there are other ways for "Illegal" to be in the ruleset -- namely, the banlist containing `"Illegal"`.

For example, Anything Goes doesn't have "Standard" in its ruleset, and does have "Illegal" in its banlist to denote that illegal Pokemon are not allowed.

I'm unsure about the specifics, but it might be possible to check `format.banlistTable['illegal']` when you get format data from `getFormat`.